server/tests/test_server_pages.py

```diff
@@ -54,12 +54,19 @@ def test_index_view(client, two_test_notebooks, fake_user, logged_in):
     }
 
 
-@pytest.mark.parametrize("username", ["testuser", "test-user", "testuser@foo.com"])
-def test_user_view_with_different_names(transactional_db, client, username):
+@pytest.mark.parametrize(
+    ("username", "first_name", "last_name"),
+    [
+        ("testuser", "User", "McUserTons"),
+        ("testuser@foo.com", "Test", ""),
+        ("test-user", "", "User"),
```

Comment:
You need to add a test case where get_full_name() returns "". I believe that would be:

`("test-user", "", "")`

I'm pretty sure that will just work thanks to your changes, but try it out and see!